src/test/java/api/loans/anonymization/AnonymizeLoansImmediatelyAPITests.java

```diff
@@ -0,0 +1,224 @@
+package api.loans.anonymization;
+
+import static api.support.matchers.LoanMatchers.isAnonymized;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.junit.MatcherAssert.assertThat;
+
+import java.net.MalformedURLException;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import api.support.builders.CheckOutByBarcodeRequestBuilder;
+import api.support.builders.LoanHistoryConfigurationBuilder;
+import org.folio.circulation.support.http.client.IndividualResource;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeUtils;
+import org.junit.Test;
+
+public class AnonymizeLoansImmediatelyAPITests extends LoanAnonymizationTests {
+
+  /**
+   * Scenario 1
+   *
+   *     Given:
+   *         An Anonymize closed loans setting of "Immediately after loan closes"
+   *         An Anonymize closed loans with associated fees/fines setting of "Immediately after fee/fine closes"
+   *         An open loan with an associated fee/fine
+   *     When the item in the loan is checked in
+   *     Then do not anonymize the loan
+   */
+  @Test
+  public void testClosedLoanWithOpenFeesAndFinesIsNotAnonymizedForSettingsOfImmediately()
```

Comment:
The scenario and the body of the test uses an open loan, this name does not seem to fit with those.

I think that reflects there are actually multiple checks here: whether the loan is closed and whether any fee or fine accounts are closed.

(Personally, I don't like the scenario comments above each test. I think they can easily get out of sync with the tests and they encourage not improving the expression of the tests themselves. I understand that other folks do, so I'm not asking for them to be deleted, only consistent)

Reply:
The here is checked in before calling anonymization.

Reply:
Test names have been updated